src/frontend/overlay/src/components/organisms/widgets/Queue2.js

```diff
@@ -0,0 +1,219 @@
+`;
+
+const rowExpand = (fullHeight) => keyframes`
+  from {
+    max-height: 0;
+  }
+
+  to {
+    max-height: ${fullHeight}px;
+  }
+`;
+
+const rowContract = (fullHeight) => keyframes`
+  from {
+    max-height: ${fullHeight}px;
+  }
+
+  to {
+    max-height: 0;
+  }
+`;
+
+const contractionStatesFeatured = (fullHeight) => ({
+    entering: {},
+    entered: { animation: rowExpand(fullHeight) },
+    exiting: { animation: rowContract(fullHeight) },
+    exited: {},
+});
+
+const contractionStates = (fullHeight) => ({
+    entering: { animation: rowExpand(fullHeight) },
+    entered: {},
+    exiting: { animation: rowContract(fullHeight) },
+    exited: {},
+});
+
+const useQueueRowsData = ({
+    width,
+    height,
+    basicZIndex = QUEUE_BASIC_ZINDEX,
+}) => {
+    const isNotShownYet = useWithTimeoutAfterRender(300);
+    const featuredRunHeight = width / QUEUE_FEATURED_RUN_ASPECT;
+
+    const { queue, totalQueueItems } = useSelector(state => state.queue);
+
+    const [isFeaturedRunMediaLoaded, setIsFeaturedRunMediaLoaded] = useState(false);
+
+    const featuredRunsRow = [];
+    const ftsRunsRows = [];
+    const regularRunsRows = [];
+    queue.forEach((run, runIndex) => {
+        const row = {
+            ...run,
+            isEven: (totalQueueItems - runIndex) % 2 === 0,
+            zIndex: basicZIndex + runIndex,
+            bottom: -QUEUE_ROW_HEIGHT2,
+            isFeatured: false,
+            isFeaturedShown: false,
+            isFts: run.result?.isFirstToSolveRun ?? false,
+        };
+        if (run.featuredRunMedia && featuredRunsRow.length === 0) {
+            row.isFeatured = true;
+            row.isFeaturedShown = isFeaturedRunMediaLoaded;
+            row.setIsFeaturedRunMediaLoaded = setIsFeaturedRunMediaLoaded;
+            row.featuredRunMediaHeight = featuredRunHeight;
+            if (isFeaturedRunMediaLoaded) {
+                featuredRunsRow.push(row);
+            } else {
+                if (run.result?.isFirstToSolveRun) {
+                    ftsRunsRows.push(row);
+                } else {
+                    regularRunsRows.push(row);
+                }
+            }
+        } else if (run.result?.isFirstToSolveRun) {
+            ftsRunsRows.push(row);
+        } else {
+            regularRunsRows.push(row);
+        }
+    });
+    if (isNotShownYet) {
+        return [...featuredRunsRow, ...ftsRunsRows, ...regularRunsRows];
+    }
+    featuredRunsRow.forEach((row) => {
+        row.zIndex = basicZIndex + MAX_QUEUE_ROWS_COUNT;
+        row.isEven = false;
+        row.bottom = height - featuredRunHeight - QUEUE_ROW_HEIGHT2;
+    });
+    ftsRunsRows.forEach((row, rowIndex) => {
+        row.bottom = QUEUE_ROW_HEIGHT2 * (rowIndex + regularRunsRows.length + 1 * (regularRunsRows.length > 0));
+    });
+    regularRunsRows.forEach((row, rowIndex) => {
+        row.bottom = QUEUE_ROW_HEIGHT2 * rowIndex;
+    });
+    return [...featuredRunsRow, ...ftsRunsRows, ...regularRunsRows];
```

Comment:
With such an approach - where a single list item corresponds to it's position on screen we are losing animation when a row is moving between sections. The elements in the DOM should not be moved. Return them in the order of the rows in the original "queue" array which should not shuffle